src/Api/Entity/AbstractEntity.php

```diff
@@ -17,7 +17,7 @@
 /**
  * @author Marc Weistroff <marc.weistroff@sensiolabs.com>
  */
-abstract class AbstractEntity implements \ArrayAccess, \Serializable
+abstract class AbstractEntity implements \ArrayAccess
```

Comment:
removing Serializable means existing payload won't be unserializable anymore
we should just add the new magic methods for now

Reply:
I've checked this. As part of this PR, we rename `serialize()` and `unserialize()` methods to `__serialize ()` and ``__unserialize()` ... so I think we should be good.

Reply:
Nope it won't: existing payloads won't be unserializable if we remove Serializable.
Unless there are no existing payloads to upgrade to the new format?

Reply:
Could this be fixed if we implement `__wakeup()` calling `$this->__unserialize()` and `__sleep()` calling `$this->__serialize()` ?

Reply:
From https://wiki.php.net/rfc/phase_out_serializable:
> If a class implements both Serializable and __serialize()/__unserialize(), the latter take precedence (on versions that support them), and the Serializable interface is only used to decode existing serialization payload using the obsolete C format. To migrate to the new mechanism, it's possible to either replace Serializable entirely (if support for PHP 7.3 and below is not needed) or to implement both (if it is needed). 

it is expected that we keep the Serializable implementation if we want to preserve compat with its serialization format.

Reply:
OK, I've readded `\Serializable`. Thanks!